homeassistant/components/islamic_prayer_times/coordinator.py

```diff
@@ -70,8 +71,8 @@ def school(self) -> str:
     def get_new_prayer_times(self) -> dict[str, Any]:
         """Fetch prayer times for today."""
         calc = PrayerTimesCalculator(
-            latitude=self.hass.config.latitude,
-            longitude=self.hass.config.longitude,
+            latitude=self.config_entry.data[CONF_LATITUDE],
+            longitude=self.config_entry.data[CONF_LONGITUDE],
```

Comment:
I think the coordinator would benefit from setting a lot of the properties in the constructor instead of accessing them every update

Reply:
done